rpc/src/server/service.rs

```diff
@@ -668,6 +670,41 @@ pub(crate) fn get_rolls_owner_current_from_context(level: &str, list: ContextLis
 
 }
 
+pub(crate) fn get_votes_listings(_chain_id: &str, block_id: &str, persistent_storage: &PersistentStorage, context_list: ContextList, state: &RpcCollectedStateRef) -> Result<Option<Vec<VoteListings>>, failure::Error> {
+    let mut listings = Vec::<VoteListings>::new();
+
+    // get block level first
+    let block_level: i64 = match get_level_by_block_id(block_id, persistent_storage, state)? {
+        Some(val) => val.try_into()?,
+        None => bail!("Block level not found")
+    };
+
+    // get the whole context
+    let ctxt = get_context(&block_level.to_string(), context_list)?;
+
+    // filter out the listings data
+    let listings_data: ContextMap = ctxt.unwrap().into_iter()
+        .filter(|(k, _)| k.contains(&"data/votes/listings/"))
```

Comment:
If it is possible, use `starts_wtih` instead of `contains`. (Don't remember if it is prefix or not).
EDIT: Just notice in further comment it should be an prefix.